library/common.h

```diff
@@ -149,4 +150,34 @@ inline void mbedtls_xor(unsigned char *r, const unsigned char *a, const unsigned
 #endif
 /* *INDENT-ON* */
 
+/* Always provide a static assert macro, so it can be used unconditionally.
+ * Note that it will expand to nothing on some systems.
+ * Can be used outside functions (but don't add a trailing ';' in that case:
+ * the semicolon is included here to avoid triggering -Wextra-semi when
+ * MBEDTLS_STATIC_ASSERT() expands to nothing).
+ * Can't use the C11-style `defined(static_assert)` on FreeBSD, since it
+ * defines static_assert even with -std=c99, but then complains about it.
+ */
+#if defined(static_assert) && !defined(__FreeBSD__)
```

Comment:
In C23, [`static_assert` will no longer be a macro](https://en.cppreference.com/w/c/error/static_assert), it's a keyword. Should we add `|| __STDC_VERSION__ >= 202300L`? Or just make a note to add this when C23 is finalized?

Reply:
We've [already decided not to pre-add stuff for C23 on other PRs](https://github.com/Mbed-TLS/mbedtls/pull/7173#issuecomment-1449871894), so let's not add the `__STDC_VERSION__` check. Not sure of the benefit of adding a `TODO` into the code, vs coming back once C23 is a real thing